sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadIT.java

```diff
@@ -138,6 +145,18 @@ public void testE2EBigtableSegmentRead() {
                     .withMaxBufferElementCount(10))
             .apply(Count.globally());
     PAssert.thatSingleton(count).isEqualTo(numRows);
-    p.run();
+    PipelineResult r = p.run();
+    checkLineageSourceMetric(r, tableId);
+  }
+
+  private void checkLineageSourceMetric(PipelineResult r, String tableId) {
```

Comment:
one cannot check Lineage metrics in unit test in BigtableIO as what BigQueryIO does is because the metrics are emitted in ServiceImpl, where unit tests used a fake Reader + ServiceImpl which does not emit Lineage (and it does not make sense add Lineage metrics there as well, because it is completely differerent code path from the real Reader / ServiceImpl)